website/src/content/docs/formatter/index.mdx

```diff
@@ -93,7 +93,138 @@ const expr =
+Biome's parser is intentionally stricter than the Prettier parser.
+It correctly identifies the following syntax errors:
+
+- A function cannot have duplicate modifiers
+- invalid order of properties' modifiers
+- Function declarations are not allowed to have bodies
+- non-abstract classes cannot have abstract properties
+- An optional chain cannot be assigned
+- The `const` modifier cannot be set on a type parameter of an interface
+- top-level return
+- etc.
+
+In Prettier, these errors aren't considered parse errors, and the AST is still built "correctly" with the appropriate nodes.
+When formatting, Prettier then just treats these nodes as normal and formats them accordingly.
+
+In Biome, the parsing errors result in `Bogus` nodes, which may contain any number of valid nodes, invalid nodes, and/or raw tokens.
+When formatting, Biome treats bogus nodes as effectively plain text, printing them out verbatim into the resulting code without any formatting, since attempting to format could be incorrect and cause semantic changes.
+
+For class properties, Prettier's current parsing strategy also uses boolean fields for modifiers, meaning only one of each kind of modifier can ever be present (accessibility modifiers are stored as a single string).
+When printing, Prettier just looks at the list of booleans and decides which modifiers to print out again. Biome instead keeps a list of modifiers, meaning duplicates are kept around and able to be analyzed (hence the parsing error messages about duplicate modifiers and ordering).
+When printing out the bogus nodes, this list is still kept intact, and printing out the unformatted text results in those modifiers continuing to exist.
+
+There are ways that Biome can address this.
+One possibility is to try to interpret the Bogus nodes when formatting and construct valid nodes out of them.
+If a valid node can be built, then it would just format that node like normal, otherwise it prints the bogus text verbatim as it does currently.
+However, this is messy and introduces a form of parsing logic into the formatter that is really not meaningful.
+
+Another option instead is to introduce some form of "syntactically-valid bogus node" into the parser, which accepts these kinds of errors that are purely semantic (duplicate modifiers, abstract properties in non-abstract classes).
+It would continue to build the nodes like normal (effectively matching the behavior in Prettier), but store them inside of a the new kind of bogus node, including the diagnostics along with it.
+When formatting, these special bogus nodes would just attempt to format the inner node, and then fallback if there's an error (the existing `format_or_verbatim` utility would do this already).
+This keeps the parsing and formatting logic separate from each other, but introduce more complexity to the parser, allowing invalid states to be considered semi-valid.
```

Comment:
```suggestion
For class properties, Prettier's current parsing strategy also uses boolean fields for modifiers, meaning only one of each kind of modifier can ever be present (accessibility modifiers are stored as a single string).
When printing, Prettier looks at the list of booleans and decides which modifiers to print out again. Biome instead keeps a list of modifiers, meaning duplicates are kept around and can be analyzed (hence the parsing error messages about duplicate modifiers and ordering).
When printing out the bogus nodes, this list is kept intact, and printing out the unformatted text results in those modifiers continuing to exist.

There are ways that Biome can address this.
One possibility is to try to interpret the Bogus nodes when formatting and construct valid nodes out of them.
If a valid node can be built, then it would just format that node like normal, otherwise, it prints the bogus text verbatim as it does currently.
However, this is messy and introduces a form of parsing logic into the formatter that is not meaningful.

Another option is to introduce some form of "syntactically-valid bogus node" into the parser, which accepts these kinds of purely semantic errors (duplicate modifiers, abstract properties in non-abstract classes).

It would continue to build the nodes like normal (effectively matching the behavior in Prettier) but store them inside of a new kind of bogus node, including the diagnostics along with it.
When formatting, these particular bogus nodes would just attempt to format the inner node and then fallback if there's an error (the existing `format_or_verbatim` utility would do this already).
This keeps the parsing and formatting logic separate from each other but introduces more complexity to the parser, allowing invalid states to be considered semi-valid.
```